### PR TITLE
fix:hang for serverUpdate failed

### PR DESCRIPTION
**Pull request by jiuker**
fix: hang for serverUpdate failed
related https://github.com/minio/operator/issues/1645

### CODE REVIEW

File: pkg/controller/main-controller.go [hunk 1]
```diff
@@ -1118,44 +1118,47 @@ func (c *Controller) syncHandler(key string) error {
...
-			if err != nil {
-				klog.Errorf("Unsupported release tag on new image, server updated but might leave dangling console deployment %v", err)
+				// Update failed, nothing needs to be changed in the container
 				return err
 			}
-			consoleDeployment, err := c.deploymentLister.Deployments(tenant.Namespace).Get(tenant.ConsoleDeploymentName())
-			if unifiedConsoleReleaseTime.Before(newVer) && consoleDeployment != nil && err == nil {
-				if err := c.deleteOldConsoleDeployment(ctx, tenant, consoleDeployment.Name); err != nil {
+		}
+		if err == nil {
```
Comment (shtripat) [inline pkg/controller/main-controller.go:1133]: This if condition not needed as we are returning from `if err != nil` part.

Comment (jiuker) [reply pkg/controller/main-controller.go:1130]: No, if err is reject by minio-server which is started with 'MINIO_UPDATE=off', we ignore it.

File: pkg/controller/main-controller.go [hunk 2]
```diff
@@ -1118,49 +1115,47 @@ func (c *Controller) syncHandler(key string) error {
...
-
-			err = fmt.Errorf("Tenant '%s' MinIO update failed with %w", tenantName, err)
-			if _, terr := c.updateTenantStatus(ctx, tenant, err.Error(), totalAvailableReplicas); terr != nil {
-				return terr
-			}
-
-			// Update failed, nothing needs to be changed in the container
-			return err
-		}
+			if err.(madmin.ErrorResponse).Code != "MethodNotAllowed" {
```
Comment (harshavardhana) [inline pkg/controller/main-controller.go:1118]:
```suggestion
			if err.(madmin.ErrorResponse).Code != "MethodNotAllowed" {
```

Use `madmin.ToErrorResponse(err)`

File: pkg/controller/main-controller.go [hunk 3]
```diff
@@ -1118,49 +1115,47 @@ func (c *Controller) syncHandler(key string) error {
...
+				err = fmt.Errorf("Tenant '%s' MinIO update failed with %w", tenantName, err)
```
Comment (harshavardhana) [inline pkg/controller/main-controller.go:1119]:
```suggestion
				err = fmt.Errorf("Tenant '%s' MinIO update failed with %w", tenantName, err)
```

Do not need to overwrite top level `err` with a custom message just to update tenantStatus() instead use `fmt.Sprintf()`

File: pkg/controller/main-controller.go [hunk 4]
```diff
@@ -1118,49 +1115,47 @@ func (c *Controller) syncHandler(key string) error {
...
-			newVer, err := miniov2.ReleaseTagToReleaseTime(us.UpdatedVersion)
-			if err != nil {
-				klog.Errorf("Unsupported release tag on new image, server updated but might leave dangling console deployment %v", err)
+				// Update failed, nothing needs to be changed in the container
 				return err
 			}
-			consoleDeployment, err := c.deploymentLister.Deployments(tenant.Namespace).Get(tenant.ConsoleDeploymentName())
-			if unifiedConsoleReleaseTime.Before(newVer) && consoleDeployment != nil && err == nil {
-				if err := c.deleteOldConsoleDeployment(ctx, tenant, consoleDeployment.Name); err != nil {
+			c.RegisterEvent(ctx, tenant, corev1.EventTypeWarning, "Set MINIO_UPDATE 'off', shouldn't update the image online!", fmt.Sprintf("Tenant %s", tenant.Name))
```
Comment (harshavardhana) [inline pkg/controller/main-controller.go:1127]:
```suggestion
			c.RegisterEvent(ctx, tenant, corev1.EventTypeWarning, "Inplace update is disabled, falling back to performing only statefulset update.", fmt.Sprintf("Tenant %s", tenant.Name))
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: pkg/controller/main-controller.go [hunk 5]
```diff
@@ -1091,19 +1091,16 @@ func (c *Controller) syncHandler(key string) error {
 
 		latest, err := c.fetchArtifacts(tenant)
```
Comment (harshavardhana) [inline pkg/controller/main-controller.go:1092]:
```suggestion
		latest, err := c.fetchArtifacts(tenant)
		defer c.removeArtifacts()
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Review by harshavardhana — DISMISSED** [review]

**Review by shtripat — APPROVED** [review]
lgtm

**Comment by pjuarezd** [comment]
checks had 20 hours running, I've restarted the tests, let's see if was a one-off fail, or somenthing to look at.

**Review by harshavardhana — APPROVED** [review]

**Review by pjuarezd — APPROVED** [review]

**Pull request merged**